docs/contract.md

```diff
@@ -9,26 +21,47 @@ The `/etc/login.gov` directory will exist on deployed instanes of login.gov apps
 | `/etc/login.gov/info/env` | `Identity::Hostdata.env` | `"int"` |
 | `/etc/login.gov/info/domain` | `Identity::Hostdata.domain` | `"login.gov"` |
 
-## EC2 instance metadata
+### Configuration via env vars
+
+To configure the app with env vars, add the following environment variables:
+
+- `LOGIN_DATACENTER`: The value `true` to indicate the IDP is in production in a deployed environment
+- `LOGIN_ENV`: The environment the app is running in (e.g. `int`, `staging`, or `prod`)
+- `LOGIN_DOMAIN`: The domain of the current app (e.g. `idp.int.identitysandbox.gov`)
+- `LOGIN_HOST_ROLE`: The role of the current host, (e.g. `idp`, `workder`, `pivcac`)
+- `LOGIN_HOST_CONFIG`: Host specific configurations (A template is available here: [https://github.com/18F/identity-devops/blob/main/kitchen/environments/environment.json.template](https://github.com/18F/identity-devops/blob/main/kitchen/environments/environment.json.template))
+
+## AWS host information
+
+Apps may need to know about the AWS environment they are configured to run in or alongside.
+If the app is running on an EC2 instance in AWS, it can read the EC2 metadata to determine the region or account ID.
+If the app is not running on an EC2 instance in AWs, it can read configurations from ENV vars
+
+### EC2 instance metadata
 
 We use [instance metadata][instance-metadata] (an HTTP request from an EC2 box) to populate basic information about our instances in EC2.
 
 - region
 - account ID
 
-The [Identity::Hostdata::EC2](../lib/identity/hostdata/ec2.rb) class helps load and read this data:
+The [Identity::Hostdata](../lib/identity/hostdata/hostdata.rb) module helps load and read this data:
 
 ```ruby
-ec2_data = Identity::Hostdata::EC2.load
-# => #<Identity::Hostdata::EC2:0x00007fc49dd30f40 ...>
-ec2_data.region
+Identity::Hostdata.aws_region
```

Comment:
This is a change that we'll need to apply to the other apps when we upgrade the gem there.